uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/AbsenceType.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+use CRM_HRLeaveAndAbsences_BAO_AbsenceType as AbsenceType;
+use CRM_HRLeaveAndAbsences_BAO_AbsencePeriod as AbsencePeriod;
+use CRM_HRLeaveAndAbsences_BAO_TOILRequest as TOILRequest;
+
+class CRM_HRLeaveAndAbsences_Service_AbsenceType {
+
+  /**
+   * Actions that occurs when an AbsenceType is updated
+   *
+   * @param \CRM_HRLeaveAndAbsences_BAO_AbsenceType $absenceType
+   */
+  public static function postUpdateActions(AbsenceType $absenceType) {
+    $oldAbsenceType = $absenceType->oldAbsenceType;
+    if ($oldAbsenceType->allow_accruals_request == true && $absenceType->allow_accruals_request == false) {
+      self::onToilDisable($absenceType);
+    }
+  }
+
+  /**
+   * The set of actions/updates that occurs when TOIL is disabled for an AbsenceType.
+   *
+   * @param \CRM_HRLeaveAndAbsences_BAO_AbsenceType $absenceType
+   */
+  private static function onToilDisable(AbsenceType $absenceType) {
+    $currentPeriod = AbsencePeriod::getCurrentPeriod();
```

Comment:
We must delete every TOILRequests in the future and that include requests for periods also in the future. Basically, we need to delete everything with expiry_date > today.

I would change deleteAllForAbsenceType to accept a $startDate and an $endDate instead of an Absence Period